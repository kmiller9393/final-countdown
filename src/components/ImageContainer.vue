<template>
    <div class="image-container">
        <h1>NASA Astronomy Picture of The Day</h1>
        <div :onload="fetchPic()">
          <img :src="spaceImage" height="450" width="650" />
        </div>
    </div>
</template>

<script>
import { fetchSpacePic } from '../utils/apiCalls';
import key from '../assets/key';

export default {
  name: 'image-container',
  data() {
    return {
      spaceImage: ''
    };
  },
  methods: {
    fetchPic() {
      fetch(`https://api.nasa.gov/planetary/apod?api_key=${key}`)
        .then(response => response.json())
        .then(spaceData => (this.spaceImage = spaceData.url));
    }
  }
};
</script>

<style scoped>
h1 {
  color: #f7f7f7;
  font-size: 3rem;
}
</style>