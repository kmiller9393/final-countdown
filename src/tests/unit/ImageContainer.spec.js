import ImageContainer from '../../components/ImageContainer.vue';
import { shallowMount } from '@vue/test-utils';

describe('ImageContainer.vue', () => {
  it('should match the snapshot', () => {
    const wrapper = shallowMount(ImageContainer);
    expect(wrapper).toMatchSnapshot();
  });
});
