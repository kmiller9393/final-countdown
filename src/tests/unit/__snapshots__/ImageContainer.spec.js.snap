// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageContainer.vue should match the snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
